feat: add neovim package guide (#9)
SnakeskinTpl/language-tools: packages/neovim/README.md

# Snakeskin Neovim plugin

## Table of Contents

- [Introduction](#introduction)
- [Installation](#installation)
- [Setup](#setup)
- [Update](#update)

## Introduction

This plugin adds support for the Snakeskin language in Neovim.
Currently, the following features are supported:

- Syntax highlighting
- Diagnostic messages (for parsing errors)

> - Go to declaration by include derective
> - Include derective validation for paths exist

Reviewer: ```suggestion
- Go to definition for `include` directive
- `include` directive validation for paths existence
```

Only Jade-like syntax is supported.

## Installation

### [lazy.nvim](https://github.com/folke/lazy.nvim)

```lua
{
    "neovim/nvim-lspconfig",
    opts = {
        servers = {
            snakeskin_ls = {}
        }
    }
}
```

## Setup

### Manual setup

> For plugin we need globally pre-installed @snakeskin/cli because plugin run cli command

Reviewer: ```suggestion
For the plugin to work, we need to install `@snakeskin/cli` globally because the plugin runs the LSP server through a CLI subcommand:
```

      npm i -g @snakeskin/cli

### Automatic setup

> If you want the cli to be installed automatically when run nvim or open *.ss file, you should add mason tools managment plugins

Reviewer: ```suggestion
If you want the CLI to be installed automatically when running `nvim` or opening a `*.ss` file, you should add the mason package manager plugin:
```

[lazy.nvim](https://github.com/folke/lazy.nvim)

```lua
{
    "williamboman/mason.nvim",
    "williamboman/mason-lspconfig.nvim",
    {
        "neovim/nvim-lspconfig",
        opts = {
            servers = {
                snakeskin_ls = {}
            }
        }
    }
}
```

## Update

### Manual update

> If you install cli by npm, you could update lsp like other npm packages.

Reviewer: ```suggestion
If you install the CLI using npm, you can update the LSP server like other npm packages:
```
